lib/ex_doc/markdown/earmark.ex

```diff
@@ -3,13 +3,29 @@ defmodule ExDoc.Markdown.Earmark do
   ExDoc extension for the Earmark MarkDown parser.
   """
 
+  alias Makeup.Formatters.HTML.HTMLFormatter
+
   @behaviour ExDoc.Markdown
 
-  def assets(_), do: []
+  @external_resource "assets/dist/ex_doc_makeup.css"
+
+  # Callback implementations
+
+  @assets [
+    # Read the CSS from the included file.
+    # This allows us to have a custom CSS theme not included in Makeup
+    # that supports both "day mode" and "night mode".
+    {"dist/ex_doc_makeup-css.css", File.read!("assets/dist/ex_doc_makeup.css")},
+    # Get the Javascript snippet directly from Makeup.
+    # If there is any need to customize it further, we can add a "ex_doc_makeup.js" file.
+    {"dist/ex_doc_makeup-js.js", HTMLFormatter.group_highlighter_javascript()}
```

Comment:
I think the distribution files should be called `ex_doc_highlighter` because if in three years we decide to use something other than makeup, we don't want to break existing adapters. Similarly, the javascript thingy should be returned by `ExDoc.Highlighter`. The HTMLFormatter is not really public API and therefore engines should not access it.

Reply:
I agree

Reply:
And more than that, the assets (including the filenames) should be defined in the ExDoc.Highlighter module, and there should be an ExDoc.Highlighter.assets/1 function that is simply called by the behaviours. That would centralize the assets in the right place.